db/migrations/65_rework_items_permissions.sql

```diff
@@ -49,18 +49,26 @@ CREATE TABLE `permissions_generated` (
         COMMENT 'can_watch_generated as an integer (to use comparison operators)',
     `can_edit_generated_value` TINYINT(3) UNSIGNED AS (`can_edit_generated` + 0) NOT NULL
         COMMENT 'can_edit_generated as an integer (to use comparison operators)',
-    `propagate_access` enum('done', 'self','children') NOT NULL
-        COMMENT 'Used by the access rights propagation algorithm to keep track of the status of the propagation',
     PRIMARY KEY (`group_id`,`item_id`),
     INDEX `item_id` (`item_id`),
-    INDEX `propagate_access` (`propagate_access`),
     CONSTRAINT `fk_permissions_generated_group_id_groups_id` FOREIGN KEY (`group_id`) REFERENCES `groups`(`id`) ON DELETE CASCADE,
     CONSTRAINT `fk_permissions_generated_item_id_items_id` FOREIGN KEY (`item_id`) REFERENCES `items`(`id`) ON DELETE CASCADE
 ) COMMENT 'Actual permissions that the group has, considering the aggregation and the propagation' ENGINE=InnoDB DEFAULT CHARSET=utf8;
 
+CREATE TABLE `permissions_propagate` (
+     `group_id` BIGINT(20) NOT NULL,
+     `item_id` BIGINT(20) NOT NULL,
+     `propagate_access` enum('self','children') NOT NULL
+         COMMENT 'Used by the access rights propagation algorithm to keep track of the status of the propagation',
```

Comment:
The attribute name does not really mean anything, what could we use instead ? 
The comment is quite the same as the table (normal). I think it would be useful to shortly state what 'self' and 'children' means here (if not already clear by describing what this value does).

`compute_next` with comment "The next permission on which the propagation should be computed, either this row or its children (through item hierarchy)" ?

Reply:
Renamed to 'propagate_to'

Reply:
fixed